R/html_document.R

```diff
@@ -360,8 +378,22 @@ html_document <- function(toc = FALSE,
 
         # flag indicating we need extra navbar css and js
         args <- c(args, pandoc_variable_arg("navbar", "1"))
-        # variables controlling padding from navbar
-        args <- c(args, pandoc_body_padding_variable_args(theme))
+
+        # navbar padding
+        if (bootstrap_version %in% "3") {
+          # TODO: remove this approach when we drop BS3 support
+          # and use the SASS based approach instead
+          args <- c(args, pandoc_body_padding_variable_args(theme))
+        } else {
+          # The $navbar-height SASS var is set by the bootstraplib theme
+          padding <- sass::sass_file(pkg_file("rmd/h/scss/navbar-padding.scss"))
+          padding_css <- bootstraplib::bootstrap_sass(
+            padding, theme = paste0(theme, "@", bootstrap_version)
+          )
+          css_file <- tempfile(fileext = ".css")
+          write_utf8(padding_css, css_file)
+          args <- c(args, "--css", pandoc_path_arg(css_file))
```

Comment:
This won't work as `css_file` is a full path to a temporary file...

Reply:
Even for `self_contained: true` I still got the local full path hard-coded in the final rendered page,
so the published site won't be able to have the css at all.